### PR TITLE
Use @EnabledIfOidcExists instead of several Gradle test tasks

### PR DESCRIPTION
**Pull request by vlsi**
#### Summary

The idea is to keep a single `test` task only, which would make IDE UI easier: it won't ask "which of test, testGithubOidc, testManual tasks would you like to use to execute the test".

#### Release Note

NONE

#### Documentation

NONE

### CODE REVIEW

**Comment by vlsi** [comment]
A side-effect is that `./gradlew build` would attempt executing test with `oidcprovider in (ANY, MANUAL)`.
I'm inclined to add a property like `skipOidc` to allow skipping OIDC flows when testing on dev machine.

**Comment by loosebazooka** [comment]
LGTM, but I would like a way to disable manual testing locally (calling to sign in)

**Comment by vlsi** [comment]
>  disable manual testing locally

Run with `-PskipOidc=true` or add `skipOidc=true` to your `$HOME/.gradle/gradle.properties`.
Is that good enough?

PS https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties
PPS https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties

**Review by loosebazooka — APPROVED** [review]
Do our actions need an update?

**Comment by loosebazooka** [comment]
So this will work because: `id-token:write` will never be accessible to an external PR? But those test just wont run anyway because of these checks?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -31,6 +31,8 @@ jobs:
       fail-fast: false
 
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```
Comment (loosebazooka) [inline .github/workflows/ci.yaml:35]: should this be id-token:write?

Comment (vlsi) [reply .github/workflows/ci.yaml]: indeed

**Review by loosebazooka — APPROVED** [review]

**Comment by loosebazooka** [comment]
> I'm inclined to add id-token: write to the regular ci.yaml

Id tokens are scoped to workflow, so there appears to be reduced risk is leaked.

**Pull request merged**